priam/src/main/java/com/netflix/priam/health/CassandraMonitor.java

```diff
@@ -172,4 +175,19 @@ public static void setIsCassadraStarted() {
         // Setting cassandra flag to true
         isCassandraStarted.set(true);
     }
+
+    private boolean isListeningOnPort(int port) {
+        String[] cmd = {"/bin/sh", "-c", "ss -tuln | grep -c " + port};
+        String line = null;
+        try {
+            Process p = Runtime.getRuntime().exec(cmd);
+            p.waitFor();
+            BufferedReader b = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            line = b.readLine();
+            b.close();
+        } catch (Exception e) {
+            logger.warn("Exception thrown while checking if process is listening on a port ", e);
+        }
+        return line != null && Integer.parseInt(line) != 0;
```

Comment:
Finally, returning false for nothing listening on the rpc_port is a peculiar failure mode if nodetool continues to report that the thrift server is running. I very rarely recommend this but I think we would benefit from a log line at INFO explaining why we are returning false.

Maybe:

logger.info("Could not find anything listening on the rpc port {}!", config.getRpcPort());